### PR TITLE
[package] Bring back save_source_file

### PR DESCRIPTION
**Pull request by zhxchen17**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59962 [package] Bring back save_source_file**

This reverts commit 44b021d21b5681c105529881bdbaefb6d3e335f6. Issue: #59737

Differential Revision: [D29113224](https://our.internmc.facebook.com/intern/diff/D29113224)

### CODE REVIEW

**Comment by zhxchen17** [comment]
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29113224).

**Comment by zhxchen17** [comment]
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29113224).

**Comment by zhxchen17** [comment]
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29113224).

File: torch/package/package_exporter.py [hunk 1]
```diff
@@ -223,6 +223,60 @@ def __init__(
         self.patterns: Dict[GlobGroup, _PatternInfo] = {}
         self._unique_id = 0
 
+    def save_source_file(
+        self, module_name: str, file_or_directory: str, dependencies=True
+    ):
+        """Adds the local file system `file_or_directory` to the source package to provide the code
```
Comment (SplitInfinity) [inline torch/package/package_exporter.py:229]:
```suggestion
        """Adds the local file system ``file_or_directory`` to the source package to provide the code
```

File: torch/package/package_exporter.py [hunk 2]
```diff
@@ -223,6 +223,60 @@ def __init__(
...
+        for `module_name`.
```
Comment (SplitInfinity) [inline torch/package/package_exporter.py:230]:
```suggestion
        for ``module_name``.
```

File: torch/package/package_exporter.py [hunk 3]
```diff
@@ -223,6 +223,60 @@ def __init__(
...
+
+        Args:
+            module_name (str): e.g. `my_package.my_subpackage`, code will be saved to provide code for this package.
+            file_or_directory (str): the path to a file or directory of code. When a directory, all python files in the directory
+                are recursively copied using :meth:`save_source_file`. If a file is named "/__init__.py" the code is treated
+                as a package.
+            dependencies (bool, optional): If ``True``, we scan the source for dependencies.
```
Comment (SplitInfinity) [inline torch/package/package_exporter.py:237]:
```suggestion
            module_name (str): e.g. ``"my_package.my_subpackage"``, code will be saved to provide code for this package.
            file_or_directory (str): the path to a file or directory of code. When a directory, all python files in the directory
                are recursively copied using :meth:`save_source_file`. If a file is named ``"/__init__.py"`` the code is treated
                as a package.
            dependencies (bool, optional): If ``True``, we scan the source for dependencies.
```

**Review by SplitInfinity — APPROVED** [review]
Just a few comments to make sure the docstrings render well in the documentation.

Thanks for working on this!

**Comment by zhxchen17** [comment]
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29113224).

**Pull request closed without merging**